src/httpResponseMessage.ts

```diff
@@ -0,0 +1,41 @@
+import { OutgoingHttpHeaders } from "http";
+
+export class HttpResponseMessage<T> {
```

Comment:
Looking at the .NET implementation, they don't use generics for this, but rather provide an `HttpContent` abstract class that gets subclassed to specific typed like `StringContent`, `StreamContent`, MultipartContent`, etc.

May be worth a discussion even though it is more complex than simply using generics here which is the quick/easy route.